### PR TITLE
ESM .min.js

### PR DESCRIPTION
**Pull request by mwcz**
This is a change by @castastrophe, extracted from the #334 branch.  It causes the ESM builds to generate two files, `pfe-foo.js` is unminified ESM and `pfe-foo.min.js` is minified ESM.

### CODE REVIEW

**Comment by mwcz** [comment]
Sourcemaps for non-minified files are now working (mapping `foo.js` &rarr; `src/foo.js`) but minified sourcemaps are not (mapping `foo.min.js`  &rarr; `foo.js`  &rarr; `src/foo.js`).

**Comment by mwcz** [comment]
One problem is that the minified files are importing non-minified files.

For example, `pfe-avatar.min.js` is importing  `../pfelement/pfelement.js` but it needs to import `../pfelement/pfelement.min.js`.  We do a similar transform in the gulpfile `merge` task when generating the `umd.js` files.

**Comment by castastrophe** [comment]
@mwcz Do you want to make the update on the importing of non-minified files to fix the sourcemaps in this branch?  Or would you prefer to make that update in a separate branch?

**Comment by mwcz** [comment]
I'd like to fix both importing min.js and sourcemaps in this branch.  I'll try to tackle that today.

**Comment by mwcz** [comment]
This is done and ready for review!

File: elements/pfe-avatar/demo/index.html [hunk 1]
```diff
@@ -13,15 +13,16 @@
...
-    <!-- <script src="/@webcomponents/webcomponentsjs/custom-elements-es5-adapter.js"></script> -->
-    <!-- <script src="/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script> -->
-    <!-- <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.5/require.min.js"></script> -->
-    <!-- <script>require(['../../pfelement/pfelement.umd.js', '../pfe-avatar.umd.js'], function(PFElement) { -->
-    <!--   PFElement.debugLog(true); -->
-    <!--   })</script> -->
+    <script src="/@webcomponents/webcomponentsjs/custom-elements-es5-adapter.js"></script>
+    <script src="/@webcomponents/webcomponentsjs/webcomponents-loader.js"></script>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.5/require.min.js"></script>
+    <script>require(['../../pfelement/pfelement.umd.min.js', '../pfe-avatar.umd.min.js'], function(PFElement) {
```
Comment (castastrophe) [inline elements/pfe-avatar/demo/index.html]: @mwcz Do you still need this debug statement?

**Comment by mwcz** [comment]
Yeah, those should be there.  Those are the sourcemaps, without them we can't debug.  If they are ever missing, raise the alarm!

**Comment by castastrophe** [comment]
Yeah, I knew they were sourcemaps, just didn't realize we needed one for every format we compile.  Makes sense though.  Yay for file explosion.  Perhaps another potential argument for a dist/compiled directory :)

File: elements/pfelement/src/pfelement.js [hunk 2]
```diff
@@ -2,6 +2,10 @@ import { autoReveal } from "./reveal.js";
 const prefix = "pfe-";
 
 class PFElement extends HTMLElement {
+  static get version() {
+    return "$version";
```
Comment (castastrophe) [inline elements/pfelement/src/pfelement.js]: Is this related to the minification?

File: package.json [hunk 3]
```diff
@@ -77,6 +77,7 @@
     "rollup-plugin-babel": "^3.0.7",
     "rollup-plugin-commonjs": "^9.1.5",
     "rollup-plugin-node-resolve": "^3.3.0",
+    "rollup-plugin-re": "^1.0.7",
```
Comment (castastrophe) [inline package.json]: Should we remove gulp-rename?  Is that being used elsewhere?

Comment (mwcz) [reply package.json]: It's used in the gulpfiles for icon, layouts, pfelement, and cp-theme.

File: elements/pfelement/src/pfelement.js (hunk 2 above)
Comment (mwcz) [reply elements/pfelement/src/pfelement.js]: Whoops, no, I was just playing around with the gulp-plugin-re.  I'll remove it from this PR.  I would like to add the version number somewhere, either as an element property or maybe in an `@preserve` comment at the top of each file ([like](https://cdnjs.cloudflare.com/ajax/libs/lodash.js/4.17.11/lodash.min.js) [many](https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js) [libs](https://cdnjs.cloudflare.com/ajax/libs/vue/2.6.7/vue.min.js) [do](https://cdnjs.cloudflare.com/ajax/libs/react/16.8.3/umd/react.production.min.js)), so if we're ever looking at a source file we can determine what version it is.

**Comment by castastrophe** [comment]
Pending travis, this is 👍

File: elements/pfe-avatar/demo/index.html (hunk 1 above)
Comment (mwcz) [reply elements/pfe-avatar/demo/index.html]: It's not needed exactly, in fact none of the changes to that file are needed (I was using this file to test out the various bundle types), so I reverted it back to how it is in master.
